### DISCUSSION TITLE
dependencyUpdates could support minor versions only

### DISCUSSION THREAD

**Discussion by rgruber1** [ben-manes/gradle-versions-plugin#769] [Ideas]
Feedback welcome but should the dependencyUpdates task support a flag to indicate that it should only check for minor version updates?

**Comment by ben-manes** [comment]
You can do that using resolution rules to reject if a different major version. While semantic versioning is becoming the norm, before that it was Maven's versioning rules that were rarely followed. Some developers liked to invent their own versioning schemes such as even being stable while odd is pre-release. That meant any assumptions we could make are a best guess, which is why you see more generic support here and others layering on top their preferred idioms in extension plugins.

**Reply by rcolombo** [reply]
@rgruber1 I spent a good chunk of time looking for an example of how to use these resolution rules for this purpose but couldn't find anything. I eventually came up with the following working solution so I'd like to share it. @ben-manes - Does this look correct or am I missing a simpler way to do this?

The following anchors the two springboot packages to `3.0.+`
```groovy
/*
Use this map in order to anchor dependencies to a specific major, minor, or patch version.
keys are "${group}:${module}" and values are the version prefix of allowed versions
Examples:
    - "org.mockito:mockito-core": "5"     // Anchors Mockito to 5.+
    - "org.mockito:mockito-core": "5.1"   // Anchors Mockito to 5.1.+
    - "org.mockito:mockito-core": "5.1.1" // Anchors Mockito to 5.1.1
Any package not in this map will resolve to the newest available version 
 */
def anchoredVersions = [
    "org.springframework.boot:spring-boot-gradle-plugin": "3.0",
    "org.springframework.boot:spring-boot-starter-web": "3.0"
]

def getAnchoredVersion = { String group, String module ->
    return anchoredVersions["${group}:${module}"]
}

// Constraints are satisfied if the provided version has the same prefix as contained in the anchoredVersions map
def satisfiesAnchoredConstraints = { String group, String module, String newVersion ->
    def anchoredVersion = getAnchoredVersion(group, module)
    return (anchoredVersion == null || newVersion.startsWith(anchoredVersion))
}

tasks.named("dependencyUpdates").configure {
    resolutionStrategy {
        componentSelection {
            all {
                if (!satisfiesAnchoredConstraints(it.candidate.group, it.candidate.module, it.candidate.version)) {
                    reject("${it.candidate.group}:${it.candidate.module}:${it.candidate.version} does not satisfy the anchored version of ${getAnchoredVersion(it.candidate.group, it.candidate.module)}")
                }
            }
        }
    }
}
```

**Reply by ben-manes** [reply]
looks great to me!
